### PR TITLE
Added Override Notification Option to Weather Module

### LINKED ISSUE #3126: Weather module: add feature "showOutdoorTemperature"
The module already has the option to display the notification of ``INDOOR_TEMPERATURE`` by other modules. People also have outdoor sensors. So adding the functioncality for ``showOutdoorTemperature`` (assuming the notification ``OUTDOOR_TEMPERATURE`` is already existant.
This could then optionally replace or complement the outdoor temperature received by the weather provider.

After this https://github.com/splattner/MMM-MQTT-Notification could be used to send the notificaton if using MQTT.

### PR DESCRIPTION
**Pull request by NolanKingdon**
Fixes #3126 

Added the option `allowOverrideNotification` to `weather.js`. This allows the module to receive the `CURRENT_WEATHER_OVERRIDE` notification. The expected payload for this notification is a full/partial `weatherObject` that is used to supplement/replace the existing `weatherObject` returned by whichever weather provider is in use. 

No visual changes.

First time contributing - let me know if I've missed something :slightly_smiling_face:

### CODE REVIEW

File: modules/default/weather/providers/localWrapper.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
+/* global Class */
+
+/* Wrapper to enable the injection of local sensor data */
+const LocalWrapper = Class.extend({
```
Comment (rejas) [inline modules/default/weather/providers/localWrapper.js:4]: LocalWrapper doesnt sound very explaining. Can you come up with a better name?
ALso I am not quite clear on how this whole thing works. Maybe add some more context in the jsdoc?

**Review by rejas — CHANGES_REQUESTED** [review]

File: modules/default/weather/providers/localWrapper.js (hunk 1 above)
Comment (NolanKingdon) [reply modules/default/weather/providers/localWrapper.js]: Added JSDocs and tried to rename things to be more clear to their use.

**Comment by rejas** [comment]
Could you check the jsdoc warnings here: https://github.com/MichMich/MagicMirror/actions/runs/6105958927?pr=3178 @NolanKingdon ?

**Review by rejas — APPROVED** [review]

**Comment by rejas** [comment]
Thx for the work @NolanKingdon will merge it now. Could you also provide a PR to update the docs in the https://github.com/MichMich/MagicMirror-Documentation/ repo?

**Pull request merged**